src/cli/commands/why.js

```diff
@@ -199,20 +213,28 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
     }
 
     // package sizes
+    // These must be declared to avoid throwing linting errors
     let packageSize = 0;
     let directSizes = [];
     let transitiveSizes = [];
-    try {
-      packageSize = await getPackageSize(match);
-    } catch (e) {}
+
+    // If we are asked to get the size of the matches, we attempt to gather size information
+    if (flags.size) {
+      try {
+        packageSize = await getPackageSize(match);
+      } catch (e) {}
```

Comment:
Never ever do blanket try-catch blocks like this, please. Why are we swallowing the error here and below?

Reply:
I hate to give this answer, but it was like that when I got here (look at the diff).

What would be a better way to handle this error? Gracefully or do we want to hard exit?

Reply:
Ah, sorry you're right. I'm guessing it was to protect against `ENOENT` or something but now I need to dig into commit logs and PR discussions. You can leave these as they are :)